src/main/java/bot/repository/UserService.java

```diff
@@ -0,0 +1,36 @@
+package bot.repository;
+
+import bot.domain.User;
+import bot.telegram.state.BotState;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@FieldDefaults(makeFinal = true,level = AccessLevel.PRIVATE)
+@RequiredArgsConstructor
+@Slf4j
+public class UserService {
+    UserRepository repository;
+
+    public User initializeUser(String chatId) {
+        Optional<User> user = repository.findById(chatId);
+        if (user.isPresent()){
```

Comment:
I personally think that it can be done a little bit more shorter :), like this :  

`repository.findById(chatId).orElse(repository.save(new User().setState(BotState.NONE).setId(chatId)));`

Up to you to change it or not, does not matter that much) 

@temalead Also consider to change the name of the method, becuase initializeUser name can easily confuse the readers about what exactly hapening inside. You are not alwyas creating new user, so I think it will be better to call method something like `getOrCreateUserByChatId`

Reply:
Thanks!